tokio/src/sync/rwlock.rs

```diff
@@ -459,6 +459,62 @@ impl<T: ?Sized> RwLock<T> {
         }
     }
 
+    /// Blockingly locks this `RwLock` with shared read access.
+    ///
+    /// This method is intended for use cases where you
+    /// need to use this rwlock in asynchronous code as well as in synchronous code.
+    ///
+    /// Returns an RAII guard which will drop the read access of this `RwLock` when dropped.
+    ///
+    /// # Panics
+    ///
+    /// This function panics if called within an asynchronous execution context.
+    ///
+    ///   - Consider using [`spawn_blocking()`][crate::runtime::Handle::spawn_blocking]
+    ///     (or [`block_in_place()`][crate::task::block_in_place]) to call, when
+    ///     within an asynchronrous execution context, any function that may
+    ///     call one of these `blocking_` operations.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::sync::Arc;
+    /// use tokio::sync::RwLock;
+    ///
+    /// #[tokio::main]
+    /// async fn main() {
+    ///     let rwlock = Arc::new(RwLock::new(1));
+    ///     let mut write_lock = rwlock.write().await;
+    ///
+    ///     let blocking_task = tokio::task::spawn_blocking({
+    ///         let rwlock = Arc::clone(&rwlock);
+    ///         move || {
+    ///             // This shall block until the `write_lock` is released.
+    ///             let read_lock = rwlock.blocking_read();
+    ///             assert_eq!(*read_lock, 0);
+    ///         }
+    ///     });
+    ///
+    ///     *write_lock -= 1;
+    ///     drop(write_lock); // release the lock.
+    ///
+    ///     // `.read()` and `.blocking_read()` don't block each other.
+    ///     let another_read_lock = rwlock.read().await;
+    ///     assert_eq!(*another_read_lock, 0);
+    ///     drop(another_read_lock);
```

Comment:
Although the code here is correct, it is dangerous to put this phrase in the documentation. Acquiring a read lock while holding another read lock is certainly not safe because of the fairness policy.

Reply:
Ah didn't know, my bad. Thanks for pointing it out!